scrimage-tests/src/test/kotlin/com/sksamuel/scrimage/core/nio/StreamingGifWriterTest.kt

```diff
@@ -3,18 +3,24 @@
 package com.sksamuel.scrimage.core.nio
 
 import com.sksamuel.scrimage.ImmutableImage
+import com.sksamuel.scrimage.nio.AnimatedGifReader
+import com.sksamuel.scrimage.nio.ImageSource
 import com.sksamuel.scrimage.nio.StreamingGifWriter
 import io.kotest.core.spec.style.WordSpec
+import io.kotest.core.spec.style.wordSpec
 import io.kotest.matchers.shouldBe
 import org.apache.commons.io.IOUtils
 import java.awt.image.BufferedImage
 import java.io.ByteArrayOutputStream
 import java.time.Duration
 
 class StreamingGifWriterTest : WordSpec({
+   // run dispose method tests
+   include(disposeMethodTests( "/com/sksamuel/scrimage/nio/animated_birds.gif")) // tests 'doNotDispose'
```

Comment:
I would be happy to add more test gifs here if desired, but I wasn't sure how many was preferred. This is the minimum test gifs to cover the expected behavior of the different defined dispose methods.